cmd/node/factory/structs.go

```diff
@@ -580,6 +597,45 @@ func prepareGenesisBlock(args *processComponentsFactoryArgs, shardsGenesisBlocks
 	return nil
 }
 
+func newEndOfEpochTrigger(args *processComponentsFactoryArgs) (endOfEpoch.TriggerHandler, error) {
+	rounder, err := round.NewRound(
+		time.Unix(args.nodesConfig.StartTime, 0),
+		args.syncer.CurrentTime(),
+		time.Millisecond*time.Duration(args.nodesConfig.RoundDuration),
+		args.syncer)
+	if err != nil {
+		return nil, err
+	}
+
+	if args.shardCoordinator.SelfId() < args.shardCoordinator.NumberOfShards() {
+		argEndOfEpoch := &shardchain.ArgsNewShardEndOfEpochTrigger{}
+		endOfEpochTrigger, err := shardchain.NewEndOfEpochTrigger(argEndOfEpoch)
+		if err != nil {
+			return nil, errors.New("error creating new end of epoch trigger" + err.Error())
```

Comment:
use predefined errors, otherwise it is difficult to test.
check other places in this file

Reply:
There are no predefined errors in structs.go. The decision taken was, that it is easier to see why a node did not started if we have clear errors.